client-js/main/client/client-factory.js

```diff
@@ -123,8 +143,8 @@ export class AbstractClientFactory {
 export class CustomClientFactory extends AbstractClientFactory {
```

Comment:
These `...ClientFactory` classes could probably benefit from having a "private" constructor (disabled via throwing `Error`).

Reply:
As discussed, we should keep these classes extendable.